src/vivarium/framework/lifecycle.py

```diff
@@ -121,7 +198,25 @@ def __init__(self):
     def add_phase(self, phase_name: str, states: List[str], loop):
         """Add a new phase to the lifecycle.
 
-        Phases must be added in order."""
+        Phases must be added in order.
+
+        Parameters
+        ----------
+        phase_name
+            The name of the phase to add.  Phase names must be unique.
+        states
+            The list of names (in order) of the states that make up the
+            life cycle phase.  State names must be unique across the entire
+            life cycle.
+        loop
+            Whether the life cycle phase states loop.
+
+        Raises
+        ------
+        LifeCycleError
+            If the phase or state names are non-unique.
```

Comment:
hmm so the docstring for LifeCycleError says `Error raised when lifecycle ordering contracts are violated.`  - is a duplicate state name an ordering violation?